hv_networks/J1587Driver.py

```diff
@@ -331,8 +342,68 @@ def set_dpa_ports(self):
+                                 'localhost by default. RP1210 J1708 interface is available on 32bit python running '
+                                 'on windows when VDA drivers are installed and configured.')
+        parser.add_argument('--truckduck-host', default='localhost', nargs='?',
+                            help='specify the network hostname of the target truckduck. For non-localhost you will '
+                                 'need to start port redirects on the target host.')
+        parser.add_argument('--rp1210-dll', nargs='?', help='the RP1210 dll name to use. uses first DLL if omitted.')
+        parser.add_argument('--rp1210-device', nargs='?', help='the RP1210 device id to use. Default is 1.',
+                            default=1, type=int)
+        parser.add_argument('--list-rp1210', help='list the RP1210 DLLs and Devices.', action='store_true')
+
+    def parse_args(self, args):
+        if args.j1708_interface in ['truckduck_1', 'j1708']:
+            self.set_ecm_ports()
+        elif args.j1708_interface in ['truckduck_2', 'j1708_2']:
+            self.set_dpa_ports()
+        elif args.j1708_interface in ['truckduck_bbplc', 'plc']:
+            self.set_dpa_ports()
+        elif args.j1708_interface in ['RP1210', 'rp1210']:
+            if not os.name == 'nt':
+                sys.stderr.write('RP1210 only supported in 32bit python on windows\n')
+                sys.exit(1)
+            self.rp1210 = True
+
+        if args.truckduck_host:
+            # TODO print a one-liner to run the needed redirects on the truckduck
+            self.truckduck_address = args.truckduck_host
+
+        if os.name == 'nt':
+            if args.rp1210_dll:
+                self.dll_name = args.rp1210_dll
+            else:
+                self.dll_name = RP1210.getAPINames()[0]
+            if args.rp1210_device:
+                self.device_id = args.rp1210_device
+            if args.list_rp1210:
+                for dll_name in RP1210.getAPINames():
+                    sys.stderr.write(f"DLL: {dll_name}\n")
+                    config = RP1210.RP1210Config(dll_name)
+                    for device in config.getDevices():
+                        sys.stderr.write(f"Device ID: {device.getID()}\n")
+                        sys.stderr.write(f"Device Description: {device.getDescription()}\n")
+                        sys.stderr.write(f"Device Name: {device.getName()}\n")
+                        sys.stderr.write(f"Device Params: {device.getParams()}\n")
+                sys.stderr.flush()
+                sys.exit(1)
+
     def make(self):
-        return J1708Driver.J1708Driver(self.ports)
+        if self.rp1210:
+            client = RP1210.RP1210Client()
+            client.setVendor(self.dll_name)
+            device_id = client.getCurrentVendor().getProtocol("J1708").getDevices()[self.device_id - 1]
+            client.setDevice(device_id)
```

Comment:
Just a note - if you already know what your target `device_id` is from `parse_args()`, there's no need to go through the whole chain of fetching vendor -> protocol -> devices; you can just assign device_id directly:

```python
client.setDevice(self.device_id)
```
(This is assuming that `parse_args()`, which I didn't read in its entirety, is successful at setting `self.device_id = the_desired_device_id`)

Reply:
The following code might be useful for checking if `self.device_id` is compatible with J1708, however:
```python
if self.device_id in client.getCurrentVendor().getProtocol("J1708").getDevices():
   device_is_compatible_with_j1708()
else:
   device_is_not_compatible_with_j1708()
```

Reply:
thank you! I've incorporated both